src/main/java/de/tum/in/www1/artemis/config/migration/setups/localvc/gitlab/MigrationEntryGitLabToLocalVC.java

```diff
@@ -0,0 +1,321 @@
+package de.tum.in.www1.artemis.config.migration.setups.localvc.gitlab;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.jgit.api.Git;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.config.migration.setups.localvc.LocalVCMigrationEntry;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.VcsRepositoryUri;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseStudentParticipation;
+import de.tum.in.www1.artemis.domain.participation.SolutionProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.domain.participation.TemplateProgrammingExerciseParticipation;
+import de.tum.in.www1.artemis.exception.localvc.LocalVCInternalException;
+import de.tum.in.www1.artemis.repository.AuxiliaryRepositoryRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseStudentParticipationRepository;
+import de.tum.in.www1.artemis.repository.SolutionProgrammingExerciseParticipationRepository;
+import de.tum.in.www1.artemis.repository.TemplateProgrammingExerciseParticipationRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.UriService;
+import de.tum.in.www1.artemis.service.connectors.localvc.LocalVCRepositoryUri;
+import de.tum.in.www1.artemis.service.connectors.vcs.AbstractVersionControlService;
+
+@Component
+@Profile("gitlab")
+public class MigrationEntryGitLabToLocalVC extends LocalVCMigrationEntry {
+
+    @Value("${artemis.version-control.default-branch:main}")
+    private String defaultBranch;
+
+    @Value("${artemis.version-control.local-vcs-repo-path:#{null}}")
+    private String localVCBasePath;
+
+    private final UserRepository userRepository;
+
+    private final Optional<AbstractVersionControlService> sourceVersionControlService;
```

Comment:
```suggestion
    private final AbstractVersionControlService sourceVersionControlService;
```
We require this for the migration. I do not think it makes sense that this is optional.

Reply:
:heavy_check_mark: with 16e6c20b7ff02ed714ba81dae0275ffc595df203